agent/src/agentclient.rs

```diff
@@ -508,14 +488,52 @@ pub mod agentclient_error {
             source: apiclient_error::Error,
         },
 
+        #[snafu(display("Unable to operate on BottlerocketShadow for {}", error))]
+        BottlerocketShadowError { error: BottlerocketShadowRWError },
+    }
+
+    impl From<BottlerocketShadowRWError> for Error {
+        fn from(err: BottlerocketShadowRWError) -> Self {
+            Self::BottlerocketShadowError { error: err }
+        }
+    }
+
+    #[derive(Debug, Snafu)]
+    #[snafu(visibility = "pub")]
+    pub enum BottlerocketShadowRWError {
+        #[snafu(display("Unable to gather system version metadata: {}", source))]
```

Comment:
Nit: we single quote the source error in other error variants
```suggestion
        #[snafu(display("Unable to gather system version metadata: '{}'", source))]
```